configure.ac

```diff
@@ -311,7 +311,7 @@ elif test -n "$NEEDS_NSL" ; then
   LIBS="$LIBS -lnsl"
 else
   # Needed for MinGW / Windows
-  AC_SEARCH_LIBS(inet_pton, ws2_32)
+  LIBS="$LIBS -lws2_32"
```

Comment:
This breaks the build outside Windows. The AC_SEARCH_LIBS is essential.

Reply:
My fault, when I saw that comment, I had understood that it was just for that platform.
But I have to say that I'm not able to compile the original code with `AC_SEARCH_LIBS()`.
It always fails because it is trying to link with an `__stdcall` function.
This function requires that the right include files to have been included at compile time.
Otherwise it will be never found by the linker if to get the` __stdcall` qualification is not applied.
Both MSYS2/MINGW and CYGWIN/MINGW-W64 fail on my PC.
It is unclear to me how previous code could work because in the past I had seen that configure scripts tested `${host_os}` variable and added `-lws2_32` directly to bypass this issue.

Reply:
I do cross-compiling libpsl with MinGW without any issue, but don't test on Windows.
There is a Meson build arch, but I doubt that anyone tested it on WinXP.

Why not check `${host_os}` for windows and skip AC_SEARCH_LIBS in that branch ?

Reply:
I did the check on `${host_os}` in my branch and it seems to work fine.
BTW, do you think that is would be acceptable something like this:

```
else
  # Platform dependant options
  case "${host_os}" in
    # MinGW / Windows
    *mingw*)
      # Select Windows NT/2000 and later, for WSAStringToAddressW()
      CPPFLAGS="$CPPFLAGS -D_WIN32_WINNT=0x500"
      # Needed for network support
      LIBS="$LIBS -lws2_32"
      ;;
    *)
      ;;
  esac
fi
```

As you can see, I also moved the macro `_WIN32_WINNT` to `$CPPFLAGS` rather than having it declared inside the source (like VisualStudio does afterall) and it seems to have fixed both issues.